### PR TITLE
docs: Add versionadded directives for list builds and scans

### PR DESCRIPTION
**Pull request by liuverta**
## Impact and Context

For the changes from

- https://github.com/VertaAI/modeldb/pull/3749
- https://github.com/VertaAI/modeldb/pull/3756
- https://github.com/VertaAI/modeldb/pull/3757
- https://github.com/VertaAI/modeldb/pull/3758
- https://github.com/VertaAI/modeldb/pull/3759

## Risks and Area of Effect

—

## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [x] Other (explain) 

Built locally—see images in PR comments.

## Reverting
- [ ] Contains Migration - _Do Not Revert_

Revert this PR.

### CODE REVIEW

File: client/verta/verta/endpoint/build/__init__.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # -*- coding: utf-8 -*-
 
-"""Verta model builds."""
+"""Verta model builds.
+
+.. versionadded:: 0.23.0
+
+"""
```
Comment (liuverta) [inline client/verta/verta/endpoint/build/__init__.py:7]: Screen Shot 2023-04-26 at 10 34 54 AM

File: client/verta/verta/endpoint/build/_build.py [hunk 2]
```diff
@@ -11,6 +11,11 @@ class Build:
     """
     An initiated docker build process of a deployable model.
 
+    .. versionchanged:: 0.23.0
+        Moved from ``verta.endpoint.Build`` to ``verta.endpoint.build.Build``.
+    .. versionadded:: 0.23.0
+        The ``date_created`` property.
```
Comment (liuverta) [inline client/verta/verta/endpoint/build/_build.py:17]: Screen Shot 2023-04-26 at 10 38 12 AM

Comment (liuverta) [reply client/verta/verta/endpoint/build/_build.py:17]: I would've put the `date_created` one with the actual property, but I don't like this blank space we end up rendering

Screen Shot 2023-04-26 at 10 29 46 AM

File: client/verta/verta/endpoint/build/_build_scan.py [hunk 3]
```diff
@@ -47,6 +51,8 @@ class ScanResultEnum(str, Enum):
 class BuildScan:
     """A scan of a Verta model build.
 
+    .. versionadded:: 0.23.0
```
Comment (liuverta) [inline client/verta/verta/endpoint/build/_build_scan.py:54]: Screen Shot 2023-04-26 at 10 40 39 AM

File: client/verta/verta/endpoint/build/_build_scan.py [hunk 4]
```diff
@@ -29,6 +31,8 @@ class ScanProgressEnum(str, Enum):
 class ScanResultEnum(str, Enum):
     """The result of a build scan.
 
+    .. versionadded:: 0.23.0
```
Comment (liuverta) [inline client/verta/verta/endpoint/build/_build_scan.py:34]: Screen Shot 2023-04-26 at 10 41 03 AM

File: client/verta/verta/endpoint/build/_build_scan.py [hunk 5]
```diff
@@ -10,6 +10,8 @@
 class ScanProgressEnum(str, Enum):
     """The current progress of a build scan.
 
+    .. versionadded:: 0.23.0
```
Comment (liuverta) [inline client/verta/verta/endpoint/build/_build_scan.py:13]: Screen Shot 2023-04-26 at 10 41 10 AM

File: client/verta/verta/registry/entities/_modelversion.py [hunk 6]
```diff
@@ -1697,6 +1697,8 @@ def list_builds(self) -> List[Build]:
         """
         Gets this model version's past and present builds.
 
+        .. versionadded:: 0.23.0
```
Comment (liuverta) [inline client/verta/verta/registry/entities/_modelversion.py:1700]: Screen Shot 2023-04-26 at 10 41 57 AM

**Review by ghost — APPROVED** [review]

**Pull request merged**